source/models/averaged_cosine_sim_model.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Corpus averaged, cosine-similarity-based binary classification model. 
+
+June, 2019
+@author: Joshua Rubin
+"""
+
+import numpy as np
+from base_model import Model
+
+class AvgCosSimModel(Model):
+    """ Derives a model that characterizes the corpus with a single normalized
+        mean vector at initialization.  Inference is performed by cosine
+        similarity with a threshold to produce true for inconsistent/fraudulent
+        and false for consistent/authentic.
+    """
+    def __init__(self, corpus=None, embedded_corpus=None):
+        """ Requires a pre-embedded corpus passed in through <embedded_corpus>.
+        """
+        if embedded_corpus is None:
```

Comment:
Good question, @mstefferson.  There might be a better way to do this, but I'm trying to write an interface that accommodates scenarios where a model could take a raw string (e.g. TDIDF) instead of (or alternatively to) a embedded vector input.  I'd like for all models to have the same method prototypes.

Reply:
Simplifying as suggested (rather than outthinking a problem I don't have yet).